src/index.ts

```diff
@@ -56,9 +58,10 @@ export class HierarchicalTokenBucket {
     this.capacity = options.maximumCapacity;
   }
 
-  child(options: Omit<HierarchicalTokenBucketOptions, 'parent'>) {
+  child(options?: Omit<HierarchicalTokenBucketOptions, 'parent'>) {
```

Comment:
Would it make sense to allow users to specify just one or the other? 

```ts
const childBucket = parentBucket.child({ refillRate: 10 });
```

```suggestion
  child(options?: Omit<Optional<HierarchicalTokenBucketOptions>, 'parent'>) {
```

Reply:
If I do that, I have to make the properties of `HierarchicalTokenBucketOptions` optional, and then check for undefined values in the constructor... I think it's messy and not a common use case. I figure the use case of inheriting options from the parent makes sense, but if a person is going to change an option anyway, it's not that painful to specify both options in that case.

Kinda related, I could totally see a future PR with setters for `maximumCapacity` and `refillRate`, and then using those when throttling errors are received to make a "smart bucket" that modulates itself.

Reply:
I love the vision @kevincasey1222